fuels-contract/src/parameters.rs

```diff
@@ -1,10 +1,15 @@
-use fuels_core::constants::{DEFAULT_BYTE_PRICE, DEFAULT_GAS_LIMIT, DEFAULT_GAS_PRICE};
+use fuel_tx::AssetId;
+use fuels_core::constants::{
+    DEFAULT_BYTE_PRICE, DEFAULT_GAS_LIMIT, DEFAULT_GAS_PRICE, NATIVE_ASSET_ID,
+};
 
 #[derive(Debug)]
 pub struct TxParameters {
     pub gas_price: u64,
     pub gas_limit: u64,
     pub byte_price: u64,
+    pub amount: u64,
+    pub asset_id: AssetId,
```

Comment:
I wouldn't put these as tx parameters since they're more parameters for a call rather than parameters for a tx. This will come into play later for sure, if we do something like allow a builder that can push multiple calls into a single tx.

ethers-rs uses dedicated functions: https://docs.rs/ethers/latest/ethers/contract/builders/struct.ContractCall.html#method.value

Instead of a dedicated function per param, maybe something like `CallParameters` could suffice. Should also include gas to forward as an optional, but that maybe could be left for a future PR since it involves more logic.

Reply:
```Rust
 let response = instance
        .get_msg_amount()
        .tx_params(tx_params)
        .call_params(call_params)
        .call()
        .await
        .unwrap();
```

Or

```Rust
 let response = instance
        .get_msg_amount()
        .tx_params(tx_params)
        .call(Some(call_params))
        .await
        .unwrap();
```
?

Reply:
John votes for the former so as to avoid `None` all the time when using default call params.